src/snake_game/utils.cljs

```diff
@@ -1,74 +0,0 @@
-(ns snake-game.utils
-  (:require [reagent.core :as reagent :refer [atom]]))
-
-(def key-code->move
-  "Mapping from integer key code to direction vector corresponding to that key"
-  {38 [0 -1]
-   40 [0 1]
-   39 [1 0]
-   37 [-1 0]})
-
-(defn rand-free-position
-  "Fuction takes snake and board-size as arguments.,
-  and returns random position not colliding wit snake body"
-  [snake [x y]]
-  (let [snake-positions (into #{} (:body snake))
-        board-positions (for [x-pos (range x)
-                              y-pos (range y)]
-                          [x-pos y-pos])]
-    (when-let [free-positions (seq (remove snake-positions board-positions))]
-      (rand-nth free-positions))))
-
-(defn collisions
-  "Returns true if snake collision with board edges or itself (snake body) is detected"
-  [{:keys [snake board] :as db}]
-  (let [{:keys [body direction]} snake
-        [x y] board
-        border-x #{x -1}
-        border-y #{y -1}
-        future-x (+ (first direction) (first (first body)))
-        future-y (+ (second direction) (second (first body)))]
-    (or (contains? border-x future-x)
-        (contains? border-y future-y)
-        (contains? (into #{} (rest body)) [future-x future-y]))))
-
-(defn change-snake-direction
-  "Changes snake head direction, only when it's perpendicular to the old head direction"
-  [[new-x new-y] [x y]]
-  (if (or (= x new-x)
-          (= y new-y))
-    [x y]
-    [new-x new-y]))
-
-(defn move-snake
-  "Move the whole snake based on positions and directions for each snake body segments"
-  [{:keys [direction body] :as snake}]
-  (let [head-new-position (mapv #(+ %1 %2) direction (first body))]
-    (assoc snake :body (into [] (drop-last (cons head-new-position body))))))
```

Comment:
You can also use `update-in` which uses a function with the old values (of `body` in this case) and returns the new content. Similar to `swap` but you can go as deep as you need with keys (e.g.: `(update-in bla [:my :nested :key] (fn [old-value-of-key] ....))`. Just so that you know it :wink: